File: src/FungeStrategy.java
/**
 * @file FungeStrategy.java
 * Strategy for executing Funge commands.
 * @author Conlan Wesson
 */

abstract class FungeStrategy{
	protected Field field;
	protected InstructionPointer ip;
	protected StackStack stack;
	
	public FungeStrategy(Field f, InstructionPointer i, StackStack s){
		field = f;
		ip = i;
		stack = s;
	}
	
	abstract public boolean execute(char cmd);
}

File: src/FungeStrategy.cpp
/**
 * @file FungeStrategy.cpp
 * Strategy for executing Funge commands.
 * @author Conlan Wesson
 */

#include "FungeStrategy.h"
#include "FungeRunner.h"

namespace Funge {

FungeStrategy::FungeStrategy(FungeRunner& r) :
	runner(r),
	field(r.getField()),
	ip(r.getIP()),
	stack(r.getStack()),
	rd(),
	gen(rd())
{
	
}

stack_t FungeStrategy::random(stack_t min, stack_t max){
	std::uniform_int_distribution<> dis(min, max);
	return dis(gen);
}

FungeError FungeStrategy::push(stack_t n){
	stack.top().push(n);
	return ERROR_NONE;
}

}
